### DISCUSSION TITLE
ssl certificate Installation

### DISCUSSION THREAD

**Discussion by GowriShanakar** [rejetto/hfs#450] [General]
HI
I love your work HFS, Its really great.
Here is the problem I am facing, cant point to folder, I mean assigning folder by default to user. Like Home directory to user. And facing issue configuring ssl certificate. Attached screen shots. I want to configure gscloud.in point to my public IP address 122.175.45.97 through https. Obtained ssl for the domain. as in the screen shot. Kindly help.
Thanking you.
Gowri Shankar
![bad ssl key](https://github.com/rejetto/hfs/assets/90929359/6dc6f05e-2af1-4980-88e2-36b828acdfb4)
![HFS 3 UPDATE ERROR](https://github.com/rejetto/hfs/assets/90929359/aaa1f14a-9d7b-4413-b9c2-c8cc695e3d66)
![Screenshot_12](https://github.com/rejetto/hfs/assets/90929359/73e9dc17-a9f7-4d8d-9bc4-8b713d066640)

**Comment by GowriShanakar** [comment]
![Screenshot_13](https://github.com/rejetto/hfs/assets/90929359/ad2be026-61e2-453e-9f81-ef91d5ac2bcf) Home directory for the user

**Comment by rejetto** [comment]
you obtained the certificate for the domain using let's encrypt in HFS, but it generates files named **acme.cert** and **acme.key**.
HFS automatically sets the files for you the correct way, but apparently you then modified it, so now it is not working.
If you look in your configuration, you are NOT pointing to the correct file.
<img width="337" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/0656141a-aded-4f9a-9c7b-b801da809203">

verify that you still have the files I just told you, and pick them in the fields above.
If you have lost the files, generate new ones.

**Reply by GowriShanakar** [reply]
Thank you very much for your response
I want to use the certificate purchased with my domain gscloud.in
![image](https://github.com/rejetto/hfs/assets/90929359/34ce3dca-8d98-4476-9f1a-c6bfdb115885)
![image](https://github.com/rejetto/hfs/assets/90929359/039da797-d073-4810-90a4-7bc31a7e8d1e)

**Reply by rejetto** [reply]
i guess that gscloud_in.crt is the certificate, so you need to pick that for the "certificate" field.
i don't see the key file. Ask your vendor about it.

the self.* files are generated with hfs by you. Delete them.

**Reply by GowriShanakar** [reply]
ok

**Comment by rejetto** [comment]
<img width="122" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/752f3898-4554-478d-826a-a777eff9eed3">

you entered here the path on your disk, but instead you have to enter the "web" path (without the domain part).
I understand the confusion, and I will make this clearer in next version.

**Reply by GowriShanakar** [reply]
Great Thank you

**Reply by GowriShanakar** [reply]
"web" path, example lz

**Reply by rejetto** [reply]
the part of the address AFTER the domain

**Reply by GowriShanakar** [reply]
Ok Tq

**Comment by nikster89** [comment]
Just wondering - when redirecting to public IP from domain via https, will IP be visible in address bar?

**Reply by rejetto** [reply]
short answer is: no, because a "domain" doesn't redirect

**Comment by charettepa** [comment]
I have a valid wildcard (*. DOMAIN) SSL cert that is working
the webpage loads and shows valid cert

however if i try to verify the SSL integrity by going to an SSL site checker
ie. https://ssltools.godaddy.com/views/certChecker
it says the server cannot be determined and no cert could be retrieved
when I check on any other of our sites with the same cert, they are fine
i must be doing something wrong

another issue is that when i go to the admin page and click on verify internet
it says it cant reach the site, not by domain and not by ip

i am operating using port 443 and forcing https
so i have NOT forwarded port 80
could this be the issue
does 80 still need to be forwarded for verification
and are any other ports needed

**Reply by rejetto** [reply]
if the "verify" fails, it's likely that your HFS is not reachable from the internet, that would also explain the result of "certChecker".
is your hfs on port 443 ?

verify doesn't need port 80.
Letsencrypt needs port 80 during generation of the certificate.

**Reply by charettepa** [reply]
it is fully accessible from the internet.  i am connecting from the outside without issue using the domain name and have been using it for about 3 months now.  its why its so strange that it fails at resolving from the admin page and that cert cannot be verified by ssl checkers

like i said the web page loads correctly without any VPN from outside, without error and cert is presented and is valid

yes i am running HFS on 443
and i am forcing HTTPS

**Reply by charettepa** [reply]
![image](https://github.com/user-attachments/assets/9011b5e8-9040-454b-bdae-29e300e17ef5)

**Reply by charettepa** [reply]
do specific countries need to be unblocked for internet verification and SSL cert check to work?

**Reply by rejetto** [reply]
ah, that's it, you are blocking other countries 😄
yes, verification uses a server on the internet (rejetto.com) that is currently located in France.
but that's true also for other service that's failing. You are blocking it.

**Reply by rejetto** [reply]
well, on a second thought, i made it so that the "verify" is let pass during the operation, so the geo-filter should not be the problem, but just for the "verify"

**Reply by charettepa** [reply]
we have geo filtering on our firewall

**Reply by rejetto** [reply]
so, you mean not on HFS but on a separate firewall. That explains it, then.

**Reply by rejetto** [reply]
or you can just allow rejetto.com if you don't want to "open" to France.
But the "verify" is not mandatory, it's just a tool, and you don't need it if you KNOW that you are working on the internet.

**Reply by charettepa** [reply]
yes sir.  i will try to open up France or rejetto.com. could this also be the issue with SSL site checkers? I made sure to try one that was based in my country and it still says it cannot communicate. 

but the website is working and is secured

**Reply by rejetto** [reply]
i guess so
